Example/PaymentSheet Example/PaymentSheet Example/EndpointSelectorViewController.swift

```diff
@@ -0,0 +1,190 @@
+            }
+            self.selectorDelegate?.selected(endpoint: input)
+            self.navigationController?.popViewController(animated: true)
+        }
+        let cancelAction = UIAlertAction(title: "Cancel", style: .cancel)
+        alertController.addAction(submitAction)
+        alertController.addAction(cancelAction)
+        present(alertController, animated: true)
+    }
+}
+
+// MARK: - TableViewController
+extension EndpointSelectorViewController {
+    override func numberOfSections(in tableView: UITableView) -> Int {
+        guard let specs = endpointSpecs else {
+            return 2
+        }
+        return specs.endpointMap.count + 1
+    }
+
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection sectionNumber: Int) -> Int {
+        if sectionNumber == 0 {
+            return 1
+        }
+        let sectionNumberIndex = sectionNumber - 1
+        guard let specs = endpointSpecs,
+              let sortedSection = sortedSections() else {
+            return 0
+        }
+        let sectionKey = sortedSection[sectionNumberIndex].key
+        guard let endpointsInSection = specs.endpointMap[sectionKey] else {
+            return 0
+        }
+        return endpointsInSection.count
+    }
+    override func tableView(_ tableView: UITableView, titleForHeaderInSection sectionNumber: Int) -> String? {
+        if sectionNumber == 0 {
+            return "Current Endpoint"
+        }
+
+        let sectionNumberIndex = sectionNumber - 1
+        guard let sortedSection = sortedSections() else {
+            return "Loading..."
+        }
+        return sortedSection[sectionNumberIndex].key
+    }
+
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = UITableViewCell(style: .default, reuseIdentifier: "endpoint")
+        guard let endpoint = endpointFor(indexPath: indexPath) else {
+            return cell
+        }
+        cell.textLabel?.text = endpoint
+        cell.textLabel?.font = .preferredFont(forTextStyle: .caption1)
+        cell.textLabel?.numberOfLines = 0
+        return cell
+    }
+    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        guard let endpoint = endpointFor(indexPath: indexPath) else {
+            return
+        }
+        self.selectorDelegate?.selected(endpoint: endpoint)
+        self.navigationController?.popViewController(animated: true)
+    }
+}
+// MARK: Cheap Data provider
+extension EndpointSelectorViewController {
+    func loadEndpointSelector(endpoint: String) {
+        let request = URLRequest(url: URL(string: endpoint)!)
+        let session = URLSession.shared.dataTask(with: request) { data, response, error in
+            guard error == nil,
+                  let data = data,
+                  let specs = self.deserializeResponse(data: data) else {
+                print("failed")
+                return
+            }
```

Comment:
Hmm... we should probably just bail at this point, and kick users back to the original screen if this happens.

Reply:
Actually -- nevermind, we can't do that.  If people enter something manually, and the link is broken, they will never be able to tap "Set Manually"